File: __tests__/node_fs_test.ml
open Jest

let _ =

describe "content" (fun _ ->
  let open ExpectJs in

  test "non-zero length" (fun _ ->
    expect (
      match [%node __filename] with
      | None -> 0
      | Some filename -> Node.Fs.readFileSync filename `utf8 |> String.length)
      |> toBeTruthy);
);

File: lib/js/__tests__/node_fs_test.js
'use strict';

var Fs = require("fs");
var Jest = require("@glennsl/bs-jest/lib/js/src/jest.js");

describe("content", (function () {
        return Jest.test("non-zero length", (function () {
                      var match = typeof (__filename) === "undefined" ? undefined : (__filename);
                      return Jest.ExpectJs[/* toBeTruthy */28](Jest.ExpectJs[/* expect */0](match !== undefined ? Fs.readFileSync(match, "utf8").length : 0));
                    }));
      }));

/*  Not a pure module */
